packages/taiko-client/driver/preconf_server/api.go

```diff
@@ -0,0 +1,95 @@
+package preconf_server
+
+import (
+	"net/http"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/labstack/echo/v4"
+)
+
+// PreconfBlockGroupStatus represents the status of a preconfirmation block group.
+type PreconfBlockGroupStatus string
+
+// PreconfBlockGroupStatus values.
+const (
+	StatusFinalBlockGroup   PreconfBlockGroupStatus = "finalBlockGroup"
+	StatusFinalPreconfGroup PreconfBlockGroupStatus = "finalPreconfGroup"
+)
+
+// PreconfTransactionsGroup represents a preconfirmation block group.
+type PreconfTransactionsGroup struct {
+	BlockID          uint64                  `json:"blockId"`
+	ID               uint64                  `json:"groupId"`
+	TransactionsList types.Transactions      `json:"transactions"`
+	GroupStatus      PreconfBlockGroupStatus `json:"groupStatus"`
+	Signature        string                  `json:"signature"`
+
+	// Block parameters
+	Timestamp             uint64         `json:"timestamp"`
+	Random                common.Hash    `json:"prevRandao"`
+	SuggestedFeeRecipient common.Address `json:"suggestedFeeRecipient"`
+	BaseFeePerGas         uint64         `json:"baseFeePerGas"`
```

Comment:
do we need this `BaseFeePerGas`?
do we need the coinbase address, is it the same as `suggestedFeeRecipient`?

Reply:
`BaseFeePerGas`  is part of the block, we need to know it to build the block, and yes, its the `coinbase`, in geth block building params, its called `suggestedFeeRecipient`

Reply:
We do need (at least) the following fields form `BlockParamsV2`:

- address coinbase
- uint64 anchorBlockId
- uint64 timestamp

Reply:
Now the driver calculates the base fee? I think this API should not request the basefee because the AVS node doesn't know how to calculate the basefee based on the previous block.

Reply:
I think we need these parameters.

<img width="296" alt="Screenshot 2024-10-21 at 23 43 58" src="https://github.com/user-attachments/assets/e6295b05-ca01-423b-a2d5-fb1c4568f5ea">

Reply:
> We do need (at least) the following fields form `BlockParamsV2`:
> 
> * address coinbase
> * uint64 anchorBlockId
> * uint64 timestamp

I see, thought AVS would create an `anchorV2` transaction in group on their own, will add

Reply:
We need all these, but the config is global, it should be read from the contract, not from the caller (AVS node)

<img width="559" alt="Screenshot 2024-10-21 at 23 45 28" src="https://github.com/user-attachments/assets/9a90aa62-136c-41f6-8762-6b60815a8a0e">

Reply:
> We need all these, but the config is global, it should be read from the contract, not from the caller (AVS node)
> 
> <img alt="Screenshot 2024-10-21 at 23 45 28" width="559" src="https://private-user-images.githubusercontent.com/99078276/378500992-9a90aa62-136c-41f6-8762-6b60815a8a0e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1MjYwOTcsIm5iZiI6MTcyOTUyNTc5NywicGF0aCI6Ii85OTA3ODI3Ni8zNzg1MDA5OTItOWE5MGFhNjItMTM2Yy00MWY2LTg3NjItNmI2MDgxNWE4YTBlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIxVDE1NDk1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQzODk2ODAzZmIzYzg0OTIxNjRkN2Q4M2ZkN2EzNGJjMzEzMjBhMzk5ODZkODAxY2NmMTBhYzE4YjY5YWQ5YzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.vmm4I8ct8gwmUFf_Oc4LGbAKFkNn-rvosaqOtEMVRYQ">

added